src/Platform/Microsoft.Testing.Extensions.VSTestBridge/ObjectModel/RunContextAdapter.cs

```diff
@@ -62,38 +61,57 @@ public RunContextAdapter(ICommandLineOptions commandLineOptions, IRunSettings ru
     /// <inheritdoc />
     public IRunSettings? RunSettings { get; }
 
-    private static string ConvertToFullyQualifiedNameFilterString(TestNodeUid testNodeUid)
+    // We use heuristic to understand if the filter should be a TestCaseId or FullyQualifiedName.
+    // We know that in VSTest TestCaseId is a GUID and FullyQualifiedName is a string.
+    private static string CreateFilter(TestNodeUid[] testNodesUid)
     {
-        StringBuilder filterString = new("FullyQualifiedName=");
+        StringBuilder filter = new();
 
-        for (int i = 0; i < testNodeUid.Value.Length; i++)
+        for (int i = 0; i < testNodesUid.Length; i++)
         {
-            char currentChar = testNodeUid.Value[i];
-            switch (currentChar)
+            if (Guid.TryParse(testNodesUid[i].Value, out Guid guid))
```

Comment:
You force changed the UID to always be the TestCase ID so there is no need for this if I think.

Reply:
It's not clear what you meant, if id is guid we use Id filter or the else will parse as in the past or?

Reply:
Let's remove the else part and let's throw a IOE saying we expect the TestNodeUID to be a TestCase ID (guid).